### PR TITLE
Autorun: convert AutoCDAction to enum class

### PR DESCRIPTION
**Pull request by notspiff**
## Description
Convert an enum to an enum class

## Motivation and context
More type safe, more modern

## How has this been tested?
It builds

## What is the effect on users?
None

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/Autorun.cpp [hunk 1]
```diff
@@ -85,8 +85,14 @@ bool CAutorun::ExecuteAutorun(const std::string& path)
...
-  if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION) == AUTOCD_RIP &&
-      pInfo->IsAudio(1) && !CServiceBroker::GetSettingsComponent()->GetProfileManager()->GetCurrentProfile().musicLocked())
+  const auto& settings = CServiceBroker::GetSettingsComponent()->GetSettings();
+  const AutoCDAction action =
+      static_cast<AutoCDAction>(settings->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION));
+  if (action == AutoCDAction::RIP && pInfo->IsAudio(1) &&
+      !CServiceBroker::GetSettingsComponent()
+           ->GetProfileManager()
+           ->GetCurrentProfile()
+           .musicLocked())
```
Comment (garbear) [inline xbmc/Autorun.cpp:95]: How about

```c++
  const auto& settings = CServiceBroker::GetSettingsComponent()->GetSettings();
  const auto& profileManager = CServiceBroker::GetSettingsComponent()->GetProfileManager();
```

```c++
  const AutoCDAction action =
      static_cast<AutoCDAction>(settings->GetInt(CSettings::SETTING_AUDIOCDS_AUTOACTION));
  if (action == AutoCDAction::RIP && pInfo->IsAudio(1) &&
      !profileManager->GetCurrentProfile().musicLocked())
```

**Review by garbear — APPROVED** [review]
No problems that I can see.

File: xbmc/Autorun.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/Autorun.cpp]: sure. ultimately i don't put any effort into cosmetics since clang-format will find some way to mangle it in any case..

Comment (garbear) [reply xbmc/Autorun.cpp]: Even without these minor improvements your work is making a big difference in quality. Thanks!

**Pull request merged**